### DISCUSSION TITLE
[FEATURE] How can i replace a Expression

### DISCUSSION THREAD

**Discussion by JosephC666** [JSQLParser/JSqlParser#1761] [Q&A] [answered]
### Problem
I want to use JSqlParse to parse some SQL for get formula result;
For example: SELECT CASE WHEN (CASE WHEN 'abcdef' like '%b%e%' THEN 1 ELSE 2 END)>1 THEN 3 ELSE 4 END;

When I implements ExpressionVisitor and override the function visit(CaseExpression caseExpression) ,after I calculated the second CASE WHEN get result is 1 , how can i replace the Expression ' (CASE WHEN 'abcdef' like '%b%e%' THEN 1 ELSE 2 END) ' to new Expression 1。-> SELECT CASE WHEN 1>1 THEN 3 ELSE 4 END;
The JSQLParser has some Function to replace an Expression Node ? 

### Some Code
```text
public class SelectItemService  SelectItemVisitor{

...override

@Override
public void visit(SelectExpressionItem selectExpressionItem){
    Expression expression = selectExpressionItem.getExpression();
    if (expression instance CaseExpression){
        CaseExpression caseExpression = (CaseExpression) expression;
        for (WhenClause whenClause : caseExperssion.getWhenClause()){
            Expression whenExpression = whenClause.getWhenExpression();
            whenExpression.accept(new ExpressionService());
            }
        }
    }
}

public class ExpressionService implements ExpressionVisitor {

...override

    @Override
    public void visit(CaseExpression caseExpression){
        List<WhenClause> whenClauses = caseExpression.getWhenClauses();
        for (WhenClause whenClause : whenClauses){
            Expression whenExpression = whenClause.getWhenExpression();
            
            if (whenExpression instanceof LikeExpression){
                // ignore other visit(....)    
                whenExpression.accept(this);    
               String leftExp = ((LikeExpression) whenExpression).getLeftExpression().toString();
               String rightExp = ((LikeExpression) whenExpression).getRightExpression().toString();
               // here I'm assuming that these two result are the same
              // how can i replace the caseExpression to (Expression)LongValue or other Expression？
             ......
           } 
        }
    }
}
```

### Additional context
The used JSQLParser Version: 4.3

Be very grateful

**Comment by manticore-projects** [comment]
Greetings.

Your `CASE` expression has a `WHEN` clause, holding the second `CASE` Expression within a `GreaterThan` and `Parenthesis`:
![image](https://user-images.githubusercontent.com/18080123/230822899-a764e6f8-fc94-4898-a2cf-d20e0ead1eec.png)

You can simply replace that second `CASE` expression of the `Parenthesis` by any other Expression.
[You can generate that AST here.](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20MoUQMiDCAqAEkEFQChZvRzsDqAJEAcrABTxIipZXU2Z6GwDkAhgEYDGAJgKYBmjsMAEkA0iCYBSVhO4TGlWoqWZo+IgEYFy7VnChYAJi07lhACKwAlLAB8sTSYyqGAZmOK94gCzvdBM8gA3EA)

**Reply by JosephC666** [reply]
Excited to hear feedback from u!

Now,I execute `parenthesis.setExpression(Expression)` to replace it,but i need save the formula result let `Parenthesis` choose modify.
so ,  If i can directly modify the Expression or recreate node in `CaseExpression` that will be great.

**Reply by manticore-projects** [reply, marked as answer]
Both is possible via the [Java API](https://javadoc.io/doc/com.github.jsqlparser/jsqlparser/latest/index.html):
1) modify the Expression of the `Parenthesis` or
2) create any new, different Expression for `Parenthesis.setExpression(..)`

**Reply by manticore-projects** [reply]
Here is a Sample how to build any Expression or Statement using the API: https://www.manticore-projects.com/JSQLParser/usage.html#build-a-sql-statement
